src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1832,6 +1852,18 @@ protected override LocalState TopState()
                             }
                         }
                     }
+
+                    if (TryGetInstanceExtensionParameter(out ParameterSymbol extensionParameter))
+                    {
+                        if (extensionParameter.RefKind != RefKind.Out)
```

Comment:
Consider collapsing these `if` conditions into one.

Reply:
I'll leave this as-is to match surrounding code